### ISSUE TITLE
Smart fields not populating when linking from another entity

### ISSUE THREAD

**Issue by gcoombe** [ForestAdmin/forest-express#58]
For example let's say I have an entity "Contact" with a couple of smart fields. I also have a "User" entity with a foreign key pointing at a Contact. If I load the forest UI on the User list and click through to an assoicated Contact, the Contact details will not have the smart fields populated, there will be a blank space.

However if I navigate to the Contact list and load a contact it will load the smart field correctly.  

I was able to fix this locally with this change: https://github.com/gcoombe/forest-express/commit/c9b3afa2cef542cd37762b5ae202b4d86f3f3cda but haven't created a Pull request yet as I feel like I might be missing how this is supposed to work.

**Comment by arnaudbesnier** [comment]
Hey @gcoombe, thanks for this feedback! That's a major issue of the existing Smart Fields implementation. You deserve a 🎖 for this issue!

It will be one our priorities for next week sprint.

🌲🌲🌲

**Comment by louisremi** [comment]
It seems this has been fixed in 1.1.0, hasn't it?

**Comment by louisremi** [comment]
It seems this has been fixed in 1.1.0, hasn't it?

**Comment by arnaudbesnier** [comment]
Yes, you're right this is fixed in the latest release!

**Comment by louisremi** [comment]
I believe the smart field documentation could greatly benefit from an update, though :-)

**Comment by arnaudbesnier** [comment]
@louisremi We are currently working hard to improve all the documentation. You can check Smart Fields documentation here: http://doc.forestadmin.com/developers-guide/#smart-fields

Do you need more detailed explanations on some specific topics?

**Comment by louisremi** [comment]
Yep!
The has-many smartfield documentation is incomplete as far as I can tell.
After some trial and errors, I figured out that I had to create a Liana.collection for the records I wanted to display in the has-many smart field. But the trickiest part was figuring out that, unless I specified an `idField` for that collection, Forest would never ever send the requests when I clicked on that field in "Related data".
Very frustrating, but at least it works now and I'll be able to enjoy my weekend :-)

**Comment by arnaudbesnier** [comment]
I don't think that you necessarily need to define a Smart Collection for a has-many Smart Field but I am not 100% sure.
@SeyZ what do you think about that?

**Comment by louisremi** [comment]
@gcoombe Can you confirm or supplement my observations of what it takes to create a has-many smart-field?

Generally speaking, the project would greatly benefit from an example project featuring examples of smart-fields/collections/actions with sources hosted on Github (and ideally a publicly accessible backoffice).

**Comment by arnaudbesnier** [comment]
![giphy-downsized](https://cloud.githubusercontent.com/assets/664261/25617296/e8a3f6be-2f41-11e7-8427-fc3265c53bd8.gif)

**Comment by gcoombe** [comment]
+1 to an express based sample project, that would go a long way.  I've also found myself experimenting with configurations until I find one that works.

@louisremi I've never been able to get a has-many smart field to work for me due to the issue described in #59 . But I actually never tried to do so by creating a Liana.collection. Previously I had followed the example in the documentation (which it seems has been removed recently?) which did not involve creating an additional Liana.collection.
